packages/auth/src/core/util/handler.ts

```diff
@@ -107,7 +114,18 @@ export function _getRedirectUrl(
       delete paramsDict[key];
     }
   }
-  return `${getHandlerBase(auth)}?${querystring(paramsDict).slice(1)}`;
+
+  // Sets the App Check token to pass to the widget
+  const appCheckToken = await auth._getAppCheckToken();
+  const appCheckTokenFragment = appCheckToken
+    ? encodeURIComponent(FIREBASE_APP_CHECK_FRAGMENT_ID) +
+      '=' +
+      encodeURIComponent(appCheckToken)
+    : '';
```

Comment:
Optional, for clarity:

```suggestion
  const appCheckTokenFragment = appCheckToken
    ? `${FIREBASE_APP_CHECK_FRAGMENT_ID}=${encodeURIComponent(appCheckToken)}` : '';
```

Reply:
Sick, sounds good